src/Scenario.cc

```diff
@@ -345,7 +345,7 @@ void Scenario::Run()
 
   int passCount = 0;
   int totalCount = 0;
```

Comment:
Do these count at the iteration or test level? In other words, if I have a scenario with 3 iterations and 3 tests in each of them, and assuming all tests pass, will these be 9/9 or 3/3 in the results?

I believe this is currently counting at the test level, which we have been moving away from in this refactor. In general, each pass/total count should aggregate the results of the entity right below it.

Suite - Counts passed/total scenarios. 1/1 in my example.
Scenario - Counts passed/total iterations. 3/3 in my example
Iteration - Counts passed/total tests. 3/3 in my example.

There is the issue that we can have a scenario with no explicit iterations, which we can default to having 1 iteration, but then the passed/total count stops making sense.

Reply:
The `passCount` and `totalCount` variables are counters for  scenario, the `iterationPassCount` and `iterationTotalCount` variables for counters for the iteration.

If you have a scenario with 3 iterations and 3 tests in each of them then:
1. `Scenario.pass_count` == 9 and `Scenario.total_count` == 9
2. `Iteration.pass_count` == 3 and `Iteration.total_count` == 3

Reply:
Does it make sense for scenarios to count the number of passed iterations instead of total tests? This is more aligned with what we had envisioned for the UI, as we want each count to aggregate the level right below it. If as a usar you want more details you can go deeper down into the tree, say, to the iteration level, to find out how many tests failed. 

I will say that you currently are our only "customer", and the one that actually uses simulation in his day to day, so if you have a strong preference for scenarios counting all tests please let me know as this will impact the UI as well.

Reply:
I think I'm getting confused, which is my typical state of mind. Here is the hierarchy that I'm working with:
1. Suite
    1. Scenario
        1. Iteration
            1. Test
                1. Trigger
                
If that is correct, then I would expect the Scenario `pass_count` to be a sum of all the tests within all the iterations that passed. 

In a similar fashion, an Iteration's `pass_count` will be the sum of all tests with the iteration itself. 

So, lets say there are 2 Scenarios in the Suite, each scenario has 3 iterations, and each iterations has 4 tests. 

## Case 1 : Everything passes

1. Suite - `pass_count==24', `total_count==24`
    1. Scenario1 - `pass_count==12`, `total_count==12`
        1. Iteration1 - `pass_count==4`, `total_count == 4`
        1. Iteration2 - `pass_count==4`, `total_count == 4`
        1. Iteration3 - `pass_count==4`, `total_count == 4`
    1. Scenario2 - `pass_count==12`, `total_count==12`
        1. Iteration1 - `pass_count==4`, `total_count == 4`
        1. Iteration2 - `pass_count==4`, `total_count == 4`
        1. Iteration3 - `pass_count==4`, `total_count == 4`
        
## Case 2 : Every second iteration fails

1. Suite - `pass_count==16`, `total_count==24`
    1. Scenario1 - `pass_count==8`, `total_count==12`
        1. Iteration1 - `pass_count==4`, `total_count == 4`
        1. Iteration2 - `pass_count==0`, `total_count == 4`
        1. Iteration3 - `pass_count==4`, `total_count == 4`
    1. Scenario2 - `pass_count==8`, `total_count==12`
        1. Iteration1 - `pass_count==4`, `total_count == 4`
        1. Iteration2 - `pass_count==0`, `total_count == 4`
        1. Iteration3 - `pass_count==4`, `total_count == 4`

## Case 3 : The first iteration passes except for the second iteration, the second iteration fails

1. Suite - `pass_count==8`, `total_count==24`
    1. Scenario1 - `pass_count==8`, `total_count==12`
        1. Iteration1 - `pass_count==4`, `total_count == 4`
        1. Iteration2 - `pass_count==0`, `total_count == 4`
        1. Iteration3 - `pass_count==4`, `total_count == 4`
    1. Scenario2 - `pass_count==0`, `total_count==12`
        1. Iteration1 - `pass_count==0`, `total_count == 4`
        1. Iteration2 - `pass_count==0`, `total_count == 4`
        1. Iteration3 - `pass_count==0`, `total_count == 4`
        
        
Does that seem right to you?

Reply:
I've updated this PR to have total counts and level counts. I've also updated  the [restructuring PR](https://gitlab.com/ignitionrobotics/cloudsim/api/-/merge_requests/58)